commit 436fe7b8b4a5016ef1fcb32bff77bde84003e15d
Author: Dave Jones <davej@redhat.com>
Date:   Mon Jun 5 14:03:50 2006 -0400

    [CPUFREQ] Make powernow-k7 work on SMP kernels.
    Even though powernow-k7 doesn't work in SMP environments,
    it can work on an SMP configured kernel if there's only
    one CPU present, however recalibrate_cpu_khz was returning
    -EINVAL on such kernels, so we failed to init the cpufreq driver.
    
    Signed-off-by: Dave Jones <davej@redhat.com>

diff --git a/arch/i386/kernel/cpu/cpufreq/powernow-k7.c b/arch/i386/kernel/cpu/cpufreq/powernow-k7.c
index 5d2b601..694d479 100644
--- a/arch/i386/kernel/cpu/cpufreq/powernow-k7.c
+++ b/arch/i386/kernel/cpu/cpufreq/powernow-k7.c
@@ -581,10 +581,7 @@ static int __init powernow_cpu_init (str
 
 	rdmsrl (MSR_K7_FID_VID_STATUS, fidvidstatus.val);
 
-	/* recalibrate cpu_khz */
-	result = recalibrate_cpu_khz();
-	if (result)
-		return result;
+	recalibrate_cpu_khz();
 
 	fsb = (10 * cpu_khz) / fid_codes[fidvidstatus.bits.CFID];
 	if (!fsb) {
